commit 794a81c71cc119520d2830cb5028d78cfe7e7176
Author: Stefan Bader <stefan.bader@canonical.com>
Date:   Wed Mar 1 11:28:07 2017 +0100

    UBUNTU: SAUCE: bcache: Fix bcache device names
    
    When adding partition support to bcache, the name assignment was not
    updated, resulting in numbers jumping (bcache0, bcache16, bcache32...).
    Fix this by taking BCACHE_MINORS into account when assigning the disk
    name.
    
    BugLink: https://bugs.launchpad.net/bugs/1667078
    
    Fixes: b8c0d91 (bcache: partition support: add 16 minors per bcacheN device)
    Cc: <stable@vger.kernel.org> # v4.10
    Signed-off-by: Stefan Bader <stefan.bader@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/drivers/md/bcache/super.c b/drivers/md/bcache/super.c
index fc0a31b13ac4..025fcab957df 100644
--- a/drivers/md/bcache/super.c
+++ b/drivers/md/bcache/super.c
@@ -791,7 +791,8 @@ static int bcache_device_init(struct bcache_device *d, unsigned block_size,
 	}
 
 	set_capacity(d->disk, sectors);
-	snprintf(d->disk->disk_name, DISK_NAME_LEN, "bcache%i", minor);
+	snprintf(d->disk->disk_name, DISK_NAME_LEN, "bcache%i",
+		 minor / BCACHE_MINORS);
 
 	d->disk->major		= bcache_major;
 	d->disk->first_minor	= minor;
